File: Module 1/case study/BaiTap/CaseStudy 6/Customer.js
class Customer {
    constructor() {
    }

    setName(name) {
        this.name = name;
    };

    getName() {
        return this.name;
    };

    setCmnd(cmnd) {
        this.cmnd = cmnd;
    };

    getCmnd() {
        return this.cmnd;
    };

    setPhoneNumber(phoneNumber) {
        this.phoneNumber = phoneNumber;
    };

    getPhoneNumber = function () {
        return this.phoneNumber;
    };

    setEmail(email) {
        this.email = email;
    };

    getEmail() {
        return this.email;
    };

    setAddress(address) {
        this.address = address;
    };

    getAddress() {
        return this.address;
    };

    setBirthday(birthday) {
        this.birthday = birthday;
    };

    getBirthday() {
        return this.birthday;
    };

    setTypeCustomer(typeCustomer) {
        this.typeCustomer = typeCustomer;
    };

    getTypeCustomer() {
        return this.typeCustomer;
    };

    setDiscount(discount) {
        this.discount = discount;
    };

    getDiscount() {
        return this.discount;
    };

    setRentdays(rentDays) {
        this.rentDays = rentDays;
    };

    getRentDays() {
        return this.rentDays;
    };

    setTyperService(typerService) {
        this.typerService = typerService;
    };

    getTyperService() {
        return this.typerService;
    };

    setTypeRoom(typeRoom) {
        this.typeRoom = typeRoom;
    };

    getTypeRoom() {
        return this.typeRoom;
    };

    setQuantityIncluded(quantityInclud) {
        this.quantityIncluded = quantityInclud;
    };

    getQuantityIncluded() {
        return this.quantityIncluded;
    };

    totalPays() {
        let money = 0;
        if (this.getTyperService() === "Villa") {
            money = 500;
        } else if (this.getTyperService()=== "House") {
            money = 300;
        } else if (this.getTyperService() === "Room") {
            money = 200;
        }
        return this.getRentDays() * money * (1 - this.getDiscount()/ 100);
    }
}

File: module2_C06/src/furama_resort/models/Customer.java
package furama_resort.models;

public class Customer implements Comparable<Customer> {
    private String name;
    private String birthday;
    private String gender;
    private String id;
    private String numberPhone;
    private String email;
    private String typeCustomer;
    private String address;
    private Services userService;

    public Services getUserService() {
        return userService;
    }

    public void setUserService(Services userService) {
        this.userService = userService;
    }


    public Customer() {
    }

    public Customer(String name, String birthday, String gender, String id, String numberPhone, String email, String typeCustomer, String address) {
        this.name = name;
        this.birthday = birthday;
        this.gender = gender;
        this.id = id;
        this.numberPhone = numberPhone;
        this.email = email;
        this.typeCustomer = typeCustomer;
        this.address = address;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getBirthday() {
        return birthday;
    }

    public void setBirthday(String birthday) {
        this.birthday = birthday;
    }

    public String getGender() {
        return gender;
    }

    public void setGender(String gender) {
        this.gender = gender;
    }

    public String getId() {
        return id;
    }

    public void setId(String id) {
        this.id = id;
    }

    public String getNumberPhone() {
        return numberPhone;
    }

    public void setNumberPhone(String numberPhone) {
        this.numberPhone = numberPhone;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getTypeCustomer() {
        return typeCustomer;
    }

    public void setTypeCustomer(String typeCustomer) {
        this.typeCustomer = typeCustomer;
    }

    public String getAddress() {
        return address;
    }

    public void setAddress(String address) {
        this.address = address;
    }

    @Override
    public String toString() {
        return "Customer{" +
                "name='" + name + '\'' +
                ", birthday='" + birthday + '\'' +
                ", gender='" + gender + '\'' +
                ", id='" + id + '\'' +
                ", numberPhone='" + numberPhone + '\'' +
                ", email='" + email + '\'' +
                ", typeCustomer='" + typeCustomer + '\'' +
                ", address='" + address + '\'' +
                ", userService=" + userService +
                '}';
    }

    public void showInfor() {
        System.out.println("Customer{" +
                "name='" + name + '\'' +
                ", birthday='" + birthday + '\'' +
                ", gender='" + gender + '\'' +
                ", id='" + id + '\'' +
                ", numberPhone='" + numberPhone + '\'' +
                ", email='" + email + '\'' +
                ", typeCustomer='" + typeCustomer + '\'' +
                ", address='" + address + '\'' +
                '}');
    }

    @Override
    public int compareTo(Customer o1) {
        int value=name.compareTo(o1.name);
        if (value==0){
            int yearFirst = Integer.parseInt(this.birthday.split("/")[2]);
            int yearSecond = Integer.parseInt(this.birthday.split("/")[2]);
            value=yearFirst-yearSecond;
//            if (yearFirst>yearSecond){value=1;}
//            else if (yearFirst<yearSecond){value=-1;}
//            else {value=0;}
        }
        return value;
    }
}
